app/ledger.js

```diff
@@ -2093,6 +2093,19 @@ var networkConnected = underscore.debounce(() => {
  * low-level utilities
  */
 
+var muonWriter = (path, payload) => {
+  muon.file.writeImportant(path, JSON.stringify(payload, null, 2), (success) => {
+    if (!success) return console.log('write error: ' + path)
+
+    if ((quitP) && (!getSetting(settings.PAYMENTS_ENABLED)) && (getSetting(settings.SHUTDOWN_CLEAR_HISTORY))) {
```

Comment:
this is kind of weird; so just to confirm... when Brave is exiting (user chose to quit) and payments aren't enabled / history is set to clear, we destroy the file by setting quitP to true and then logging something?  I mention it's weird because the deletion only happens after the file is written to (versus just deleting the file).